src/pybind/mgr/cephadm/serve.py

```diff
@@ -1183,7 +1183,7 @@ def _write_client_files(self,
                             client_files: Dict[str, Dict[str, Tuple[int, int, int, bytes, str]]],
                             host: str) -> None:
         updated_files = False
-        if host in self.mgr.offline_hosts:
+        if host in [h.hostname for h in self.mgr.cache.get_unreachable_hosts()]:
```

Comment:
it would be nice to have a func that simply checks the unreachable states (offline, maintenance) for the targeted host (instead of building and enumerating a list of unreachable hosts).

Reply:
Took a look at this and we do this in a lot of spots so going to handle it in a follow up. I've created a tracker for it https://tracker.ceph.com/issues/61548